internal/venice-common/src/main/java/com/linkedin/venice/pushmonitor/PushStatusDecider.java

```diff
@@ -129,11 +129,18 @@ public Pair<ExecutionStatus, Optional<String>> checkPushStatusAndDetailsByPartit
             getPartitionStatus(partitionStatus, pushStatus.getReplicationFactor(), partition.getInstanceToStateMap());
 
         if (executionStatus == ERROR) {
+          Optional<String> leaderError = getLeaderErrorInfo(
```

Comment:
It is possible that there will be several ERROR leader replicas when a node goes bad, and rebalance will take some time to kick in, so this will cause the upstream caller only reset the first ERROR leader replica, right?
Here is an example:
1. P1, P2, P3 are all hosted as leader replicas by a bad node.
2. P1, P2, P3 reports error almost at the same time.
3. Controller receives P1 leader error, and reset it.
4. P2, P3 ERROR comes, and P1 leader was disabled, but it will take some time to move leadership from the bad node, and then P1 could be reset again.
5. Because of this race condition, Controller won't reset P2/P3 leader replicas.

I think we should take a holistic view of the ERROR leader replicas, and reset all of them for every notification.

Since checking every errored leader replica will need to pass a lot of info to the upstream caller, I am thinking to pass a Optional callback to this `PushStatusDecider`, so that it could react against a holistic view and Router could skip this param.

Reply:
If we could execute the callback this class, we don't need to create a function: `getLeaderErrorInfo` to duplicate the logic, but instead, let `getPartitionStatus` return more information, right?

Reply:
I see for multi-partition ERROR replicas case, it will just reset the first one, then after helix possibly elects a new good leader, it will reset the next one.  I have updated the code to reset all error replica at shot. But for stores with many partitions, will that cause a rebalance storm ? or need to throttle the disabling.

Reply:
In theory, it shouldn't cause a rebalance storm since leadership handover is different from rebootstrapping a new replica, which is much heavier.
We have the Helix node-level throttling, and I don't think we need extra throttling.